public/docs/ts/latest/guide/component-communication.jade

```diff
@@ -0,0 +1,707 @@
+
+  Now, our task is to inject the list of heroes into `HeroJobBoard` so that we can create the corresponding `HeroPanel` components. We are going to use
+  the `@ContentChildren` annotation of Angular. We decorate a component property with `@ContentChildren(C)`, and set the type of the property as `QueryList<C>`,
+  where `C` represents the type of the children. Here is how we use this decorator with `HeroJobBoard`:
+  
++makeExample('component-communication/ts/fragments/heroes-to-invite-2.ts', null, 'take-job-contentchildren/hero-job-board.ts (extract)', {blk: /(@ContentChildren\(HeroCrm\)|QueryList&lt;HeroCrm&gt;)/g, pnk: /(heroesToInvite)/g})
+
+:marked
+  When `HeroJobBoard` is instantiated, the framework assigns a `QueryList<HeroCrm>` instance to `heroesToInvite`. At this moment this list is empty.
+  `QueryList` observes the changes of the parent component's content, updates its content accordingly. Changes in `QueryList`'s content are observed
+  by change detection, and trigger re-rendering, if necessary.
+  
+  When the **Invite heroes** button is clicked, `inviteHeroes()` copies the content of `heroesToInvite` into `heroesInvited`.
+  
+.alert.is-helpful
+  :marked
+    The `inviteHeros()` method uses the `toArray()` method of `QueryList` to convert the `QueryList<HeroCrm>` object to a `Hero[]`. Because the shape of
+    `HeroCrm` is compatible with `Hero` &mdash; they have the same properties &mdash; TypeScript allows this implicit conversion.
+    
+    The component lifecycle has an explicit point when the content of `HeroJobBoard` is collected by the framework, and then `heroesToInvite` is updated. You can hook into
+    this lifecycle event by defining the `ngAfterContentInit()` method, and carry out any additional setup you may need to use the collection of children.
+
+:marked
+  We do not need to change any other components to complete our task. When the app is running, and the **Invite heroes** button is clicked, we 
+  can see 8 heroes instead of the 6 we saw in the previous scenarios:
+  
+figure.image-display
+  img(src="/resources/images/devguide/component-communication/with-contentchildren-ui.png" alt="Heroes by @ContentChildren")
+  
++makeExample('component-communication/ts/fragments/heroes-to-invite-2.ts', null, 'take-job-contentchildren/hero-job-board.ts (extract)', {blk: /(@ContentChildren\(HeroCrm\)|QueryList&lt;HeroCrm&gt;)/g, pnk: /(heroesToInvite)/g})
+
+:marked
+  The `JobBoardWithContent` component nests 8 `hero-crm` children: 6 coming from `Hero.heroes`, and 2 others (Dynama and Magma) added with direct markup.
+  
+  ### A few additional comments about `@ContentChildren`
+  
+  * `@ContentChildren` provides a mechanism for a parent to collect its children defined as content. We can use _all the communication techniques between the
+  parent and its children_ (in both direction) you already learned in this chapter.
+  * Components can have multiple `@ContentChildren` properties witch different child types.
```

Comment:
The point you're making about different content children with different types won't be understood by the reader, mostly because we don't show it. I almost didn't get it myself.

No point in talking about something we don't show. FWIW, I don't think this is the place to show or talk about it.